File: scripts/reset_bit.ts
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient();

async function main() {
    console.log("Wiping all BIT properties...");
    const res = await prisma.property.deleteMany({
        where: {
            source_provider: 'BIT'
        }
    });

    // Let's also enforce NTA source_provider for everything else, just in case (optional)
    
    console.log(`[BIT-Reset] Deleted old data | Scraped 0 new properties. Deleted ${res.count} rows.`);
}

main().catch(console.error).finally(() => prisma.$disconnect());

File: crawler/reset_bit.py
import os
import psycopg2
from dotenv import load_dotenv

load_dotenv("/Users/kimtrung/keibai-finder/web/.env")
db_url = os.environ.get("DATABASE_URL").replace("?schema=public", "")

conn = psycopg2.connect(db_url)
cur = conn.cursor()

print("Wiping all BIT properties...")
cur.execute("DELETE FROM \"Property\" WHERE source_provider = 'BIT' OR source_provider IS NULL")
deleted_rows = cur.rowcount

conn.commit()
cur.close()
conn.close()

print(f"[BIT-Reset] Deleted old data | Deleted {deleted_rows} rows.")
